### PR TITLE
load urdf/srdf from json-wrapped string input

### PR DESCRIPTION
**Pull request by ClintLiddick**
Backwards-compatible with previous interface, this adds additional ability to load a urdf and (optionally) srdf from a json-wrapped string through OpenRAVE plugin command.

### CODE REVIEW

File: src/urdf_loader.h [hunk 1]
```diff
@@ -24,16 +27,19 @@ namespace or_urdf
...
+    bool loadJsonString(std::ostream& sout, std::istream& sin);
     
     /** Constructs plugin and registers functions */
     URDFLoader(OpenRAVE::EnvironmentBasePtr env) : OpenRAVE::ModuleBase(env)
     {
       __description = "URDFLoader: Loader that imports URDF files.";
       _env = env;
 
-      RegisterCommand("load", boost::bind(&URDFLoader::load, this, _1, _2),
+      RegisterCommand("load", boost::bind(&URDFLoader::loadFile, this, _1, _2),
```
Comment (cdellin) [inline src/urdf_loader.h]: This looks great, but I think the names are now confusing. I proposed that we (a) rename `URDFLoader::load` -> `URDFLoader::loadModel`, (b) rename `URDFLoader::loadFile` -> `URDFLoader::loadURI` and rename its command registration to `LoadURI`, and (c) create a new wrapper command called `URDFLoader::load` registered as `Load` which internally forwards to `LoadURI` (perhaps eventually with a deprecation warning?). What do y'all think?

Comment (mkoval) [inline src/urdf_loader.h]: I agree with @cdellin. Lets:

1. Rename `URDFLoader::load` to `URDFLoader::loadModel`.
2. Rename `URDFLoader::loadFile` to `URDFLoader::loadURI`.
3. Rename the `Load` command to `LoadURI`.
4. Rename the `LoadJsonString` command to `LoadString`.
5. Add a `Load` command that prints a deprecation warning and forwards to `LoadURI`.

**Comment by psigen** [comment]
If we are shoving a random header-only JSON library in here, I prefer https://github.com/nlohmann/json to `picojson`.

**Comment by ClintLiddick** [comment]
@cdellin @mkoval I made those changes. @psigen that library looks good but doesn't work with 14.04 gcc version 4.8 :cry: .

I'll open a PR on herbpy to remove the deprecation warning.

**Comment by mkoval** [comment]
👍 I'll merge once Travis is happy.

**Pull request merged**